device/interface.yaml

```diff
@@ -0,0 +1,56 @@
+components:
+  schemas:
+    Interface.Active:
+      type: object
+      properties:
+        active:
+          description: >-
+            If enabled means that this part of the configuration including any
+            active 'children' nodes will be advertised to peer. 
+            If disabled, this means that though config is present, it is not
+            taking any part of the test but can be activated at run-time to
+            advertise just this part of the configuration to the peer.
+          type: boolean
+          default: true
+
+    Interface:
+      x-include: 
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      description: >-
+        A container for emulated interfaces and protocol devices.
+      type: object
+      required: [container_name]
+      properties:
+        container_name: 
+          description: |-
+            The unique name of a Port or Lag object that will contain the 
+            emulated interfaces and/or protocol devices.
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+          - '/components/schemas/Lag/properties/name' 
+        iface_type: 
+          description: |-
+            ethernet - This is a layer 2 interface defined by unique Mac address and optional vlans.
```

Comment:
is this completely accurate?  you can also put an IPv4 or IPv6 address on an Ethernet interface, no?

Reply:
yes. corrected